site/src/components/UserAutocomplete/UserAutocomplete.tsx

```diff
@@ -0,0 +1,126 @@
+import CircularProgress from "@material-ui/core/CircularProgress"
+import { makeStyles } from "@material-ui/core/styles"
+import TextField from "@material-ui/core/TextField"
+import Autocomplete from "@material-ui/lab/Autocomplete"
+import { useMachine } from "@xstate/react"
+import { User } from "api/typesGenerated"
+import { AvatarData } from "components/AvatarData/AvatarData"
+import debounce from "just-debounce-it"
+import { ChangeEvent, useEffect, useState } from "react"
+import { searchUserMachine } from "xServices/users/searchUserXService"
+
+export type UserAutocompleteProps = {
+  value?: User
+  onChange: (user: User | null) => void
+}
+
+export const UserAutocomplete: React.FC<UserAutocompleteProps> = ({ value, onChange }) => {
+  const styles = useStyles()
+  const [isAutocompleteOpen, setIsAutocompleteOpen] = useState(false)
+  const [searchState, sendSearch] = useMachine(searchUserMachine)
+  const { searchResults } = searchState.context
+  const [selectedValue, setSelectedValue] = useState<User | null>(value || null)
+
+  // seed list of options on the first page load
+  useEffect(() => {
+    const query = value ? value.email : ""
+    sendSearch("SEARCH", { query })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  // when selected value changes, update search terms
+  useEffect(() => {
+    const query = selectedValue ? selectedValue.email : ""
+    sendSearch("SEARCH", { query })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedValue])
```

Comment:
This is already done in the `handleFilterChange`. I don't think we need the useEffect here 🤔

Reply:
Hmm, this is what I'm trying to solve with this `useEffect`: 
![Kapture 2022-09-27 at 09 34 30](https://user-images.githubusercontent.com/19142439/192541026-f633172b-4ac0-4bde-affa-8b5e717f11a7.gif)

Is there a better way?

Reply:
Ahhh I see. So when you clear the input, you want to clear the results as well right? If it is, I would add a new event to the search machine called `CLEAR_RESULTS` that sets the results to undefined and do this logic in the onChange handler like:
```
....
// This happens when the user clears the results with the "x" button
if(option === null) {
    send("CLEAR_RESULTS")
}
```
Unfortunately, the MUI Autocomplete does not have an `onClear` 😔 that would be a better place. I like to put logic on event handlers because I think they are better to follow when they are called than useEffect but it is up to you. If you think this is not necessary feel free to merge it.

Reply:
That worked well! Thanks!